### PR TITLE
Firefox content fixes

### PR DESCRIPTION
**Pull request by matejak**
- Fixed Firefox remediation regressions caused by porting bash shared remediations to jinja.
- Fixed OVALs that have not been passing on RHEL7
- Added a possibility to raise exceptions from inside Jinja macros at expand-time.

As the test suite doesn't pick the Firefox product directories, one can test the PR by running profile-mode test run with the `stig` profile.

Note: Inclusion of ` . /usr/share/scap-security-guide/remediation_functions` is needed in order to populate variables.

### CODE REVIEW

File: firefox/guide/firefox/firefox_preferences-lock_settings/firefox_preferences-lock_settings_config_file/oval/firefox.xml [hunk 1]
```diff
@@ -13,11 +13,11 @@
     </criteria>
   </definition>
 
-  <ind:textfilecontent54_test check="all" comment="Check for configuration filename" id="test_firefox_preferences-config_filename" version="1">
+  <ind:textfilecontent54_test check="all" comment="Check for configuration filename" id="test_firefox_preferences-config_filename" version="2">
     <ind:object object_ref="object_firefox_preferences-config_filename" />
   </ind:textfilecontent54_test>
   <ind:textfilecontent54_object id="object_firefox_preferences-config_filename" version="1">
-    <ind:path operation="pattern match">^\/usr\/(|local\/)lib(|64)\/firefox\/defaults\/preferences</ind:path>
+    <ind:path operation="pattern match">^\/usr\/(|local\/)lib(|64)\/firefox\/defaults\/(preferences|pref)</ind:path>
```
[the remark this answers is not in the document]
Comment (matejak) [reply firefox/guide/firefox/firefox_preferences-lock_settings/firefox_preferences-lock_settings_config_file/oval/firefox.xml]: I don't know, but on RHEL7 system where I have tested the profile, Fixefox uses `prefs`. The remediation considers both `preferences` and `prefs`.

File: shared/macros-bash.jinja [hunk 2]
```diff
@@ -202,6 +202,28 @@ sed -i "s/disable.*/disable         = yes/gI" "/etc/xinetd.d/$xinetd"
 {{%- endif -%}}
 {{%- endmacro -%}}
 
+
+{{%- macro _put_into_firefox_cfg(what, key, where, sed_separator) -%}}
+# If the key exists, change it. Otherwise, add it to the config_file.
+if LC_ALL=C grep -m 1 -q '^{{{ what }}}("{{{ key }}}", ' "{{{ where }}}"; then
+  sed -i 's{{{ sed_separator }}}{{{ what }}}("{{{ key }}}".*{{{ sed_separator }}}{{{ what }}}("{{{ key }}}", '"$value)"';{{{ sed_separator }}}g' "{{{ where }}}"
+else
+  echo '{{{ what }}}("{{{ key }}}", '"$value"');' >> "{{{ where }}}"
```
[the remark this answers is not in the document]
Comment (matejak) [reply shared/macros-bash.jinja]: I have slightly restructured the code to make it less surprising, I hope it helped.

**Comment by matejak** [comment]
I have added a commit that fixes indentation, so it may be helpful to skip that one when reviewing the PR.

**Comment by jan-cerny** [comment]
It's great. The report after remediation is still the same.

**Pull request merged**